File: miraculous-frontend/miraculous-ui/src/dto/artist-link-dto/artist-link-dto.ts
export class ArtistLinkDto {

  id: number;
  login: string;
  pictureUrlName: string;
  genres: string[];
  totalNumberOfLikes: number;
  numberOfTracks: number;
  numberOfAlbums: number;


  constructor(id?: number,
              login?: string,
              pictureUrl?: string,
              genres?: string[],
              totalNumberOfLikes?: number,
              numberOfTracks?: number,
              numberOfAlbums?: number) {
    this.id = id;
    this.login = login;
    this.pictureUrlName = pictureUrl;
    this.genres = genres;
    this.totalNumberOfLikes = totalNumberOfLikes;
    this.numberOfAlbums = numberOfAlbums;
    this.numberOfTracks = numberOfTracks;
  }
}

File: miraculous-backend/src/main/java/com/example/miraculousbackend/dto/ArtistLinkDto.java
package com.example.miraculousbackend.dto;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ArtistLinkDto {

    private Long id;
    private String login;
    private String pictureUrlName;
    private List<String> genres;
    private Long totalNumberOfLikes;
    private long numberOfTracks;
    private long numberOfAlbums;

}
